packages/core/src/domain/report/reportObservable.ts

```diff
@@ -0,0 +1,153 @@
+import { mergeObservables, Observable } from '../../tools/observable'
+import { DOM_EVENT, includes, addEventListener } from '../../tools/utils'
+import { monitor } from '../internalMonitoring'
+
+export interface BrowserWindow {
+  ReportingObserver?: ReportingObserver
+}
+
+export interface ReportingObserverOption {
+  types: BrowserReportType[]
+  buffered: boolean
+}
+
+interface ReportingObserverCallback {
+  (reports: BrowserReport[], observer: ReportingObserver): void
+}
+
+export declare class ReportingObserver {
+  static readonly supportedEntryTypes: readonly string[]
+  constructor(callback: ReportingObserverCallback, option: ReportingObserverOption)
+  disconnect(): void
+  observe(): void
+  takeRecords(): PerformanceEntryList
+}
+
+export interface BrowserReport {
+  type: BrowserReportType
+  url: string
+  body: DeprecationReportBody | InterventionReportBody
+}
+
+interface DeprecationReportBody {
+  id: string
+  message: string
+  lineNumber: number
+  columnNumber: number
+  sourceFile: string
+  anticipatedRemoval?: Date
+}
+
+interface InterventionReportBody {
+  id: string
+  message: string
+  lineNumber: number
+  columnNumber: number
+  sourceFile: string
+}
+
+export const ReportType = {
+  intervention: 'intervention',
+  deprecation: 'deprecation',
+  csp_violation: 'csp_violation',
+} as const
+
+export type ReportType = typeof ReportType[keyof typeof ReportType]
+
+export interface Report {
+  type: ReportType
+  message: string
+  stack?: string
+}
+
+export type BrowserReportType = 'intervention' | 'deprecation'
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏It is a bit hard to follow what is Browser types and what is our types. 
It could be interesting to extract the browser types to a dedicated file with the actual browser names somewhere.

Reply:
I've put browser types into report/browser.type.ts